packages/http-client/src/ceramic-http-client.ts

```diff
@@ -198,15 +198,16 @@ export default class CeramicClient implements CeramicApi {
   }
 
   addDoctypeHandler<T extends Doctype>(doctypeHandler: DoctypeHandler<T>): void {
-    this._doctypeHandlers[doctypeHandler.name] = doctypeHandler
+    this._doctypeConstructors[doctypeHandler.name] = doctypeHandler.doctype
   }
 
-  findDoctypeHandler<T extends Doctype>(doctype: string): DoctypeHandler<T> {
-    const doctypeHandler = this._doctypeHandlers[doctype]
-    if (doctypeHandler == null) {
-      throw new Error(`Failed to find doctype handler for doctype ${doctype}`)
+  findDoctypeConstructor<T extends Doctype>(doctype: string) {
+    const constructor = this._doctypeConstructors[doctype]
+    if (constructor) {
+      return constructor as DoctypeConstructor<T>
+    } else {
+      throw new Error(`Failed to find doctype constructor for doctype ${doctype}`)
```

Comment:
We can just remove this method as it's not being used.

Reply:
But it is used :) See: https://github.com/ceramicnetwork/js-ceramic/pull/758/files#diff-1e1d2be8d1b9f18852db4000834507647dc77def6e5447c3b133610b44046312R131 and below.

Reply:
Right, but we can simply use a map instead. See https://github.com/ceramicnetwork/js-ceramic/pull/668

Reply:
I believe there is an implicit assumption here: one could _only_ have `tile` and `caip10-link` doctypes. Alternative formulation: it is _impossible_ to have other doctypes.

Using map/record is fine, as long as it always follows a happy path. If not, error message regarding null objects and absent methods could be quite hard to understand. This thing happens to construct meaningful error.

So, could you confirm, `caip10-link` and `tile` are the only expected doctypes, and the user have no right to add own doctypes?

Reply:
Ok, right might be a good idea to keep it because of the error being thrown. 
It's expected that users add their own doctyes.